src/image/p5.Image.js

```diff
@@ -878,21 +878,22 @@ p5.Image = class {
     }
     const currBlend = this.drawingContext.globalCompositeOperation;
 
-    let scaleFactor = 1;
+    let imgScaleFactor = this._pixelDensity;
+    let maskScaleFactor = 1;
     if (p5Image instanceof p5.Renderer) {
-      scaleFactor = p5Image._pInst._pixelDensity;
+      maskScaleFactor = p5Image._pInst._pixelDensity;
```

Comment:
`p5.Image` also has a pixel density too now, maybe we can add a branch to handle when the input is an image too? https://github.com/processing/p5.js/blob/37d3324b457b177319ed65468201f2806a66eff5/src/image/p5.Image.js#L150